src/ios/registerNativeModule.js

```diff
@@ -50,6 +54,80 @@ const addProjectToLibraries = (libraries, file) => {
 };
 
 /**
+ * Given folder, it returns an array of all header files
+ * inside it, ignoring node_modules and examples
+ */
+const getHeadersInFolder = (folder) =>
+  glob
+    .sync('**/*.h', {
+      cwd: folder,
+      nodir: true,
+      ignore: GLOB_EXCLUDE_PATTERN,
+    })
+    .map(file => path.join(folder, file));
+
+/**
+ * Given an array of headers it returns search path so Xcode can find them
+ * If there are multiple header files located accross different folders,
+ * it finds the top-most one and returns recursive version of it
+ */
+const getHeaderSearchPath = (sourceDir, headers) => {
+  const directories = union(
+    headers.map(path.dirname)
+  );
+
+  if (directories.length === 1) {
+    return `"$(SRCROOT)/${path.relative(sourceDir, directories[0])}"`;
+  }
+
+  const directory = directories.reduce((topMostDir, currentDir) => {
+    const currentFolders = currentDir.split(path.sep);
+    const topMostFolders = topMostDir.split(path.sep);
+
+    if (currentFolders.length === topMostFolders.length) {
+      return currentFolders.slice(0, -1).join(path.sep);
+    }
+
+    return currentFolders.length < topMostFolders.length
+      ? currentDir
+      : topMostDir;
+  });
```

Comment:
Otherwise, in case:
````
src/Base/A.h
src/B.h
```
arrays will be:
['users', 'grabbou', 'project', 'node_modules', 'react-module', 'src', 'base']
['users', 'grabbou', 'project', 'node_modules', 'react-module', 'src'];

We return the one that's shorter.